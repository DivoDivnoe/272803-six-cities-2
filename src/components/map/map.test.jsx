import React from 'react';
import renderer from 'react-test-renderer';
import Map from './map.jsx';

describe(`Map component`, () => {
  it(`is rendered correctly`, () => {
    const coords = {
      city: {
        latitude: 0,
        longitude: 0,
        zoom: 1
      },
      hotels: [
        {
          latitude: 10,
          longitude: 10,
          zoom: 1
        }
      ]
    };

    const leaflet = jest.genMockFromModule(`leaflet`);
    leaflet.map = () => {
      return ({
        setView: jest.fn()
      });
    };
    leaflet.icon = jest.fn();
    leaflet.tileLayer = () => {
      return {addTo: jest.fn()};
    };
    leaflet.marker = () => {
      return {
        on: jest.fn(),
        addTo: jest.fn()
      };
    };

    const activeHotel = 0;
    const city = ``;

    const tree = renderer.create(
        <Map
          coords={coords}
          leaflet={leaflet}
          activeHotel={activeHotel}
          city={city}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
