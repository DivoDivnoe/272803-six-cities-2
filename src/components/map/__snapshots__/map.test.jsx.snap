// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component is rendered correctly 1`] = `
<div
  id="map"
  style={
    Object {
      "height": "100%",
    }
  }
/>
`;
